### PR TITLE
feat: handle namespace resource selector change CRPS

### PR DESCRIPTION
**Pull request by Arvindthiru**
### Description of your changes

Fixes #

I have:

- [ ] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/controllers/placement/controller.go [hunk 1]
```diff
@@ -142,6 +142,19 @@ func (r *Reconciler) handleUpdate(ctx context.Context, placementObj fleetv1beta1
 		}
 	}
 
+	// Validate namespace selector consistency for NamespaceAccessible CRPs.
+	if isNamespaceAccessibleCRP(placementObj) {
+		isValid, err := r.validateNamespaceSelectorConsistency(ctx, placementObj)
+		if err != nil {
+			klog.V(2).InfoS("Namespace selector validation failed, stopping reconciliation", "placement", placementKObj, "error", err)
```
Comment (jwtty) [inline pkg/controllers/placement/controller.go]: Log as error?

File: pkg/controllers/placement/controller.go [hunk 2]
```diff
@@ -160,16 +173,18 @@ func (r *Reconciler) handleUpdate(ctx context.Context, placementObj fleetv1beta1
...
-		}
-
+		// First update CRP status with the schedule condition
 		if updateErr := r.Client.Status().Update(ctx, placementObj); updateErr != nil {
 			klog.ErrorS(updateErr, "Failed to update the status", "placement", placementKObj)
 			return ctrl.Result{}, controller.NewUpdateIgnoreConflictError(updateErr)
 		}
+		klog.V(2).InfoS("Updated the placement status with scheduled condition", "placement", placementKObj)
+
+		if isNamespaceAccessible := isNamespaceAccessibleCRP(placementObj); isNamespaceAccessible {
```
Comment (jwtty) [inline pkg/controllers/placement/controller.go]:
```suggestion
		if isNamespaceAccessibleCRP(placementObj) {
```

File: pkg/controllers/placement/controller.go [hunk 3]
```diff
@@ -212,18 +227,18 @@ func (r *Reconciler) handleUpdate(ctx context.Context, placementObj fleetv1beta1
...
-		return ctrl.Result{}, err
-	}
-
 	if err := r.Client.Status().Update(ctx, placementObj); err != nil {
 		klog.ErrorS(err, "Failed to update the status", "placement", placementKObj)
 		return ctrl.Result{}, err
 	}
 	klog.V(2).InfoS("Updated the placement status", "placement", placementKObj)
 
+	if isNamespaceAccessible := isNamespaceAccessibleCRP(placementObj); isNamespaceAccessible {
```
Comment (jwtty) [inline pkg/controllers/placement/controller.go]:
```suggestion
	if isNamespaceAccessibleCRP(placementObj) {
```

File: pkg/controllers/placement/controller.go [hunk 4]
```diff
@@ -1275,3 +1290,34 @@ func emitPlacementStatusMetric(placementObj fleetv1beta1.PlacementObj) {
...
+		return metav1.Condition{
+			Type:               string(fleetv1beta1.ClusterResourcePlacementStatusSyncedConditionType),
+			Status:             metav1.ConditionUnknown,
+			Reason:             condition.InvalidResourceSelectorsReason,
+			Message:            "NamespaceAccessible ClusterResourcePlacement doesn't specify a resource selector which selects a namespace",
+			ObservedGeneration: generation,
+		}
+	}
+	// Determine condition based on sync result
+	if syncErr != nil {
```
Comment (jwtty) [inline pkg/controllers/placement/controller.go]: You can put this check first? Then check if targetNamespace == nil as the one line 1296?

File: pkg/controllers/placement/controller.go [hunk 5]
```diff
@@ -1275,3 +1290,34 @@ func emitPlacementStatusMetric(placementObj fleetv1beta1.PlacementObj) {
 	// This condition is used solely for metric reporting purposes.
 	metrics.FleetPlacementStatusLastTimeStampSeconds.WithLabelValues(placementObj.GetNamespace(), placementObj.GetName(), strconv.FormatInt(placementObj.GetGeneration(), 10), "Completed", string(metav1.ConditionTrue), "Completed").SetToCurrentTime()
 }
+
+// buildStatusSyncedCondition creates a StatusSynced condition based on the sync result.
+func buildStatusSyncedCondition(generation int64, targetNamespace string, syncErr error) metav1.Condition {
```
Comment (jwtty) [inline pkg/controllers/placement/controller.go]: Why don't we move this function to namesapce_status_sync.go? As it's not used in controller.go at all.

File: pkg/controllers/placement/namespace_status_sync.go [hunk 6]
```diff
@@ -27,81 +27,225 @@ import (
...
+
+	return true
+}
+
+// filterStatusSyncedCondition creates a copy of the placement status with the
+// ClusterResourcePlacementStatusSynced condition removed. This is used when
+// creating ClusterResourcePlacementStatus objects because the StatusSynced
+// condition is only relevant for the main CRP object, not the CRPS object.
+// Returns a filtered copy of the status with all conditions except StatusSynced.
+func filterStatusSyncedCondition(status placementv1beta1.PlacementStatus) placementv1beta1.PlacementStatus {
```
Comment (jwtty) [inline pkg/controllers/placement/namespace_status_sync.go]: We could use pointers as the argument and return value?

File: pkg/controllers/placement/namespace_status_sync.go [hunk 7]
```diff
@@ -27,81 +27,225 @@ import (
...
+	placementKObj := klog.KObj(&placementv1beta1.ClusterApprovalRequest{})
+	crp, _ := placementObj.(*placementv1beta1.ClusterResourcePlacement)
+
+	// Extract target namespace from resource selectors.
+	targetNamespace := extractNamespaceFromResourceSelectors(crp)
 	if targetNamespace == "" {
-		// Not NamespaceAccessible or no namespace found - skip sync.
-		klog.V(2).InfoS("Skipped processing CRP to create/update ClusterResourcePlacementStatus", "crp", klog.KObj(crp))
-		return nil
+		klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("no namespace selector found despite NamespaceAccessible scope")),
```
Comment (jwtty) [inline pkg/controllers/placement/namespace_status_sync.go]: Thinking twice, this should be a userError?

File: pkg/controllers/placement/namespace_status_sync.go [hunk 8]
```diff
@@ -27,81 +27,225 @@ import (
...
+	klog.V(2).InfoS("Successfully handled ClusterResourcePlacementStatus", "crp", placementKObj, "namespace", targetNamespace, "operation", op)
+	return targetNamespace, nil
+}
+
+// handleNamespaceAccessibleCRP handles the complete workflow for ClusterResourcePlacements
+// with NamespaceAccessible scope. It syncs the ClusterResourcePlacementStatus object in
+// the target namespace, builds a StatusSynced condition based on the sync result, adds
+// the condition to the CRP, and updates the CRP status.
+func (r *Reconciler) handleNamespaceAccessibleCRP(ctx context.Context, placementObj placementv1beta1.PlacementObj) error {
+	// Sync ClusterResourcePlacementStatus object if StatusReportingScope is NamespaceAccessible
```
Comment (jwtty) [inline pkg/controllers/placement/namespace_status_sync.go]:
```suggestion
	// Sync ClusterResourcePlacementStatus object if StatusReportingScope is NamespaceAccessible.
```

File: pkg/controllers/placement/namespace_status_sync.go [hunk 9]
```diff
@@ -27,81 +27,225 @@ import (
...
+func (r *Reconciler) setSyncedConditionAndUpdateStatus(ctx context.Context, placementObj placementv1beta1.PlacementObj, condition metav1.Condition) error {
+	placementKObj := klog.KObj(placementObj)
+
+	// Set the condition on the placement object.
+	placementObj.SetConditions(condition)
+
+	// Update the placement status.
+	if err := r.Client.Status().Update(ctx, placementObj); err != nil {
+		klog.ErrorS(err, "Failed to update the placement status with StatusSynced condition", "crp", placementKObj)
+		return err
```
Comment (jwtty) [inline pkg/controllers/placement/namespace_status_sync.go]: wrap with IgnoreConflict?

File: pkg/controllers/placement/namespace_status_sync.go [hunk 10]
```diff
@@ -27,81 +27,225 @@ import (
...
 }
+
+// validateNamespaceSelectorConsistency validates that ClusterResourcePlacements with
+// NamespaceAccessible scope have consistent namespace selectors. This function assumes
+// the placement is already confirmed to be NamespaceAccessible. It checks if the
+// namespace selector has changed from what was originally selected and sets the
+// StatusSynced condition accordingly. Returns an error if validation fails or if
+// there are issues updating the placement status.
+func (r *Reconciler) validateNamespaceSelectorConsistency(ctx context.Context, placementObj placementv1beta1.PlacementObj) (bool, error) {
+	placementKObj := klog.KObj(&placementv1beta1.ClusterApprovalRequest{})
```
Comment (jwtty) [inline pkg/controllers/placement/namespace_status_sync.go]: ?

File: pkg/controllers/placement/namespace_status_sync.go [hunk 11]
```diff
@@ -27,81 +27,225 @@ import (
...
+	crp, _ := placementObj.(*placementv1beta1.ClusterResourcePlacement)
+
+	// Extract target namespace from resource selectors.
+	targetNamespaceFromSelector := extractNamespaceFromResourceSelectors(crp)
+	if targetNamespaceFromSelector == "" {
+		klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("no namespace selector found despite NamespaceAccessible scope")),
```
Comment (jwtty) [inline pkg/controllers/placement/namespace_status_sync.go]: NewUserError?

File: test/e2e/actuals_test.go [hunk 12]
```diff
@@ -1520,54 +1525,141 @@ func customizedPlacementStatusUpdatedActual(
...
+		wantStatus := buildWantCRPStatus(placementKey, placement.GetGeneration(), wantSelectedResourceIdentifiers, wantSelectedClusters, wantUnselectedClusters, wantObservedResourceIndex, resourceIsTrackable)
+		if diff := cmp.Diff(placement.GetPlacementStatus(), wantStatus, placementStatusCmpOptions...); diff != "" {
+			return fmt.Errorf("Placement status diff (-got, +want): %s", diff)
 		}
+		return nil
+	}
+}
 
-		var wantPlacementConditions []metav1.Condition
+func buildWantCRPStatus(
```
Comment (jwtty) [inline test/e2e/actuals_test.go]:
```suggestion
func buildWantPlacementStatus(
```

File: test/e2e/actuals_test.go [hunk 13]
```diff
@@ -1520,54 +1525,141 @@ func customizedPlacementStatusUpdatedActual(
...
+	}
+
+	if len(wantUnselectedClusters) > 0 {
 		if len(wantSelectedClusters) > 0 {
-			wantPlacementConditions = placementRolloutCompletedConditions(placementKey, placement.GetGeneration(), false)
+			wantPlacementConditions = placementSchedulePartiallyFailedConditions(placementKey, placementGeneration)
 		} else {
-			// We don't set the remaining resource conditions.
-			wantPlacementConditions = placementScheduledConditions(placementKey, placement.GetGeneration())
+			// we don't set the remaining resource conditions if there is no clusters to select
```
Comment (jwtty) [inline test/e2e/actuals_test.go:1570]:
```suggestion
			// we don't set the remaining resource conditions if there is no clusters to select.
```

File: test/e2e/actuals_test.go [hunk 14]
```diff
@@ -1520,54 +1525,141 @@ func customizedPlacementStatusUpdatedActual(
...
-		wantStatus := &placementv1beta1.PlacementStatus{
-			Conditions:                  wantPlacementConditions,
-			PerClusterPlacementStatuses: wantPlacementStatus,
-			SelectedResources:           wantSelectedResourceIdentifiers,
-			ObservedResourceIndex:       wantObservedResourceIndex,
+// customizedPlacementStatusUpdatedActualWithStatusSynced extends customizedPlacementStatusUpdatedActual
+// to optionally include the ClusterResourcePlacementStatusSyncedConditionType condition.
+// This is useful for integration tests that need to verify the StatusSynced condition without modifying
+// the widely-used customizedPlacementStatusUpdatedActual function.
+func customizedPlacementStatusUpdatedActualWithStatusSynced(
```
Comment (jwtty) [inline test/e2e/actuals_test.go]: This function should be CRP only.

File: test/e2e/placement_status_sync_test.go [hunk 15]
```diff
@@ -124,11 +123,16 @@ var _ = Describe("ClusterResourcePlacementStatus E2E Tests", Ordered, func() {
...
 
-			// Verify the recreated CRPS matches the current CRP status
+			// Verify the recreated CRPS matches the current CRP status.
 			crpsMatchesActual := statussyncutils.CRPSStatusMatchesCRPActual(ctx, hubClient, crpName, appNamespace().Name)
 			Eventually(crpsMatchesActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Recreated ClusterResourcePlacementStatus should match current CRP status")
 		})
 
+		It("CRP status should remain unchanged", func() {
+			statusUpdatedActual := namespaceAccessibleCRPStatusUpdatedActual(workResourceIdentifiers(), allMemberClusterNames, nil, "0", metav1.ConditionTrue)
+			Eventually(statusUpdatedActual, crpsEventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP status with 3 clusters")
```
Comment (jwtty) [inline test/e2e/placement_status_sync_test.go]: Use Consistently?

File: test/e2e/placement_status_sync_test.go [hunk 16]
```diff
@@ -236,12 +238,13 @@ var _ = Describe("ClusterResourcePlacementStatus E2E Tests", Ordered, func() {
 			ensureCRPAndRelatedResourcesDeleted(crpName, allMemberClusters)
 		})
 
-		It("should create ClusterResourcePlacementStatus in the work resources namespace", func() {
+		It("should update CRP status with 2 clusters as expected", func() {
```
Comment (jwtty) [inline test/e2e/placement_status_sync_test.go]: comment is not correct?

File: test/e2e/placement_status_sync_test.go [hunk 17]
```diff
@@ -266,46 +269,156 @@ var _ = Describe("ClusterResourcePlacementStatus E2E Tests", Ordered, func() {
...
+
+	Context("NamespaceAccessible CRP with namespace selector change for consistency validation - namespace exists", func() {
+		crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+
+		BeforeAll(func() {
+			// Create work resources that will be selected by the CRP.
+			createWorkResources()
+
+			// Create another namespace.
+			ns := corev1.Namespace{
```
Comment (jwtty) [inline test/e2e/placement_status_sync_test.go:307]: Also verify this namespace is not placed?

File: test/e2e/actuals_test.go [hunk 18]
```diff
@@ -1520,54 +1525,141 @@ func customizedPlacementStatusUpdatedActual(
...
+	resourceIsTrackable bool,
+	statusSyncedConditionStatus metav1.ConditionStatus,
+) *placementv1beta1.PlacementStatus {
+	var wantStatus *placementv1beta1.PlacementStatus
+	// Start with the base status
+	if statusSyncedConditionStatus == metav1.ConditionUnknown {
+		// Only StatusSynced condition should get updated in this scenario, since namespace selector has changed.
+		wantStatus = buildWantCRPStatus(placementKey, placementGeneration-1, wantSelectedResourceIdentifiers, wantSelectedClusters, wantUnselectedClusters, wantObservedResourceIndex, resourceIsTrackable)
+	} else {
+		wantStatus = buildWantCRPStatus(placementKey, placementGeneration, wantSelectedResourceIdentifiers, wantSelectedClusters, wantUnselectedClusters, wantObservedResourceIndex, true)
```
Comment (jwtty) [inline test/e2e/actuals_test.go]:
```suggestion
		wantStatus = buildWantCRPStatus(placementKey, placementGeneration, wantSelectedResourceIdentifiers, wantSelectedClusters, wantUnselectedClusters, wantObservedResourceIndex, resourceIsTrackable)
```

File: test/e2e/placement_status_sync_test.go (hunk 17 above)
Comment (Arvindthiru) [reply test/e2e/placement_status_sync_test.go:307]: That should be verified when we check CRP status, it's a fundamental functionality of CRP which should be covered by existing tests as well

File: pkg/controllers/placement/controller_integration_test.go [hunk 19]
```diff
@@ -2189,26 +2189,31 @@ var _ = Describe("Test ClusterResourcePlacement Controller", func() {
...
 					Finalizers: []string{placementv1beta1.PlacementCleanupFinalizer},
 				},
 				Spec: crp.Spec,
+				Status: placementv1beta1.PlacementStatus{
+					ObservedResourceIndex: "0",
+					Conditions: []metav1.Condition{
+						{
+							Status: metav1.ConditionUnknown,
+							Type:   string(placementv1beta1.ClusterResourcePlacementScheduledConditionType),
+							Reason: condition.SchedulingUnknownReason,
```
Comment (jwtty) [inline pkg/controllers/placement/controller_integration_test.go:2205]: Just curious: why do we have this SchedulingUnknown condition?

File: pkg/controllers/placement/controller_integration_test.go [hunk 20]
```diff
@@ -2303,20 +2308,25 @@ var _ = Describe("Test ClusterResourcePlacement Controller", func() {
 				Spec: crp.Spec,
 				Status: placementv1beta1.PlacementStatus{
 					Conditions: []metav1.Condition{
+						// ObservedResourceIndes is not set because resource selector is invalid.
```
Comment (jwtty) [inline pkg/controllers/placement/controller_integration_test.go]:
```suggestion
						// ObservedResourceIndex is not set because resource selector is invalid.
```

File: pkg/controllers/placement/namespace_status_sync.go [hunk 21]
```diff
@@ -27,81 +27,257 @@ import (
...
+
+	return true
+}
+
+// filterStatusSyncedCondition creates a copy of the placement status with the
+// ClusterResourcePlacementStatusSynced condition removed. This is used when
+// creating ClusterResourcePlacementStatus objects because the StatusSynced
+// condition is only relevant for the main CRP object, not the CRPS object.
+// Returns a filtered copy of the status with all conditions except StatusSynced.
+func filterStatusSyncedCondition(status *placementv1beta1.PlacementStatus) placementv1beta1.PlacementStatus {
```
Comment (jwtty) [inline pkg/controllers/placement/namespace_status_sync.go]: Any reason we keep the return value as the obj instead of a pointer?

File: pkg/controllers/placement/namespace_status_sync.go (hunk 10 above)
Comment (jwtty) [reply pkg/controllers/placement/namespace_status_sync.go]: Why is this marked as resolved? What's ClusterApprovalRequest?

File: pkg/controllers/placement/namespace_status_sync_test.go [hunk 22]
```diff
@@ -42,328 +41,495 @@ var (
...
+
+					// Additional validation that LastUpdatedTime is set (ignored in comparison).
+					if gotCRPS.LastUpdatedTime.IsZero() {
+						t.Fatal("Expected LastUpdatedTime to be set on CRPS")
+					}
+				}
+			}
+
+			// Check StatusSynced condition on the original CRP using cmp.Diff.
+			if tc.wantCRPCondition != nil {
```
Comment (jwtty) [inline pkg/controllers/placement/namespace_status_sync_test.go]: I don't think we need this check? We just put the wanted condition in test case and compare with what we got? In this case, it works for nil condition too.

File: pkg/controllers/placement/namespace_status_sync_test.go [hunk 23]
```diff
@@ -42,328 +41,495 @@ var (
...
+			// Check if error expectation matches.
+			if tc.wantError && err == nil {
+				t.Fatal("Expected error but got none")
+			}
+			if !tc.wantError && err != nil {
+				t.Fatalf("handleNamespaceAccessibleCRP() failed: %v", err)
+			}
+
+			// Check ClusterResourcePlacementStatus creation/update using cmp.Diff.
+			if tc.wantCRPSOperation && tc.targetNamespace != "" {
```
Comment (jwtty) [inline pkg/controllers/placement/namespace_status_sync_test.go:240]: What about the not-creating case?

File: pkg/controllers/placement/controller_integration_test.go (hunk 19 above)
Comment (Arvindthiru) [reply pkg/controllers/placement/controller_integration_test.go:2205]: I think it's an initial condition set on the CRP before scheduling occurs, I used this as reference https://github.com/kubefleet-dev/kubefleet/blob/main/pkg/controllers/placement/controller_integration_test.go#L2048

File: apis/placement/v1beta1/clusterresourceplacement_types.go [hunk 24]
```diff
@@ -1287,6 +1287,19 @@ const (
...
+	// StatusReportingScope is NamespaceAccessible.
+	//
+	// It can have the following condition statuses:
+	// * True: Fleet has successfully created or updated the ClusterResourcePlacementStatus object
+	//   in the target namespace.
+	// * False: Fleet has failed to create or update the ClusterResourcePlacementStatus object
+	//   in the target namespace.
+	// * Unknown: Fleet cannot determine the status synchronization state due to invalid resource
+	//   selectors (e.g., no namespace selector found, or namespace selector inconsistency detected).
+	ClusterResourcePlacementStatusSyncedConditionType ClusterResourcePlacementConditionType = "ClusterResourcePlacementStatusSynced"
```
Comment (ryanzhang-oss) [inline apis/placement/v1beta1/clusterresourceplacement_types.go:1300]: what happens to the CRP whose StatusReportingScope is ClusterScopeOnly? Do we set this condition as "Unknown" or not set this?

File: pkg/controllers/placement/namespace_status_sync.go [hunk 25]
```diff
@@ -27,81 +27,257 @@ import (
...
+// isNamespaceAccessibleCRP checks if the given placement object is a ClusterResourcePlacement
+// with StatusReportingScope set to NamespaceAccessible. Returns true if the placement is
+// a CRP with NamespaceAccessible scope, false otherwise.
+func isNamespaceAccessibleCRP(placementObj placementv1beta1.PlacementObj) bool {
 	crp, ok := placementObj.(*placementv1beta1.ClusterResourcePlacement)
 	if !ok {
-		// This is a ResourcePlacement, not a ClusterResourcePlacement - skip sync.
-		klog.V(2).InfoS("Skipped processing RP to create/update ClusterResourcePlacementStatus")
-		return nil
+		return false
```
Comment (ryanzhang-oss) [inline pkg/controllers/placement/namespace_status_sync.go:70]: is there any reason to remove the log? Is it too verbose?

File: pkg/controllers/placement/namespace_status_sync.go [hunk 26]
```diff
@@ -27,81 +27,257 @@ import (
...
+	}
+
+	// Only process if StatusReportingScope is NamespaceAccessible.
+	if crp.Spec.StatusReportingScope != placementv1beta1.NamespaceAccessible {
+		klog.V(2).InfoS("StatusReportingScope is not NamespaceAccessible", "crp", klog.KObj(placementObj))
```
Comment (ryanzhang-oss) [inline pkg/controllers/placement/namespace_status_sync.go:75]: why add this log but remove the previous one?

File: apis/placement/v1beta1/clusterresourceplacement_types.go (hunk 24 above)
Comment (ryanzhang-oss) [reply apis/placement/v1beta1/clusterresourceplacement_types.go:1300]: when there is namespace selected but the StatusReportingScope is Namespace, it seems that this condition is actually false as we know we didn't create the ClusterResourcePlacementStatus object

File: test/e2e/placement_status_sync_test.go (hunk 17 above)
Comment (ryanzhang-oss) [inline test/e2e/placement_status_sync_test.go:307]: Where is this Ns used?

File: test/e2e/placement_status_sync_test.go [hunk 27]
```diff
@@ -266,46 +269,156 @@ var _ = Describe("ClusterResourcePlacementStatus E2E Tests", Ordered, func() {
...
 					return err
 				}
-				// Change number of clusters to 1 to trigger status update.
-				crp.Spec.Policy.NumberOfClusters = ptr.To(int32(3))
+				crp.Spec.ResourceSelectors = []placementv1beta1.ResourceSelectorTerm{
+					{
+						Group:   corev1.GroupName,
+						Version: "v1",
+						Kind:    "Namespace",
+						Name:    "random-namespace",
```
Comment (ryanzhang-oss) [inline test/e2e/placement_status_sync_test.go]: AFAIK, this ns does not exist in the test?

File: apis/placement/v1beta1/clusterresourceplacement_types.go (hunk 24 above)
Comment (Arvindthiru) [reply apis/placement/v1beta1/clusterresourceplacement_types.go:1300]: Currently when StatusReportingScope is ClusterScopeOnly we don't set the ClusterResourcePlacementStatusSynced condition on the CRP

I don't get the second comment, is the suggestion to set ClusterResourcePlacementStatusSynced condition to false on the CRP when StatusReportingScope is ClusterScopeOnly ?

File: pkg/controllers/placement/namespace_status_sync.go [hunk 28]
```diff
@@ -27,81 +27,268 @@ import (
...
+			filteredConditions = append(filteredConditions, condition)
+		}
+	}
+	filteredStatus.Conditions = filteredConditions
+
+	return filteredStatus
+}
+
+// buildStatusSyncedCondition creates a StatusSynced condition based on the sync result.
+func buildNamespaceAccessibleCondition(generation int64, isSyncError, isResourceSelectorInvalid bool, msg string) metav1.Condition {
```
Comment (ryanzhang-oss) [inline pkg/controllers/placement/namespace_status_sync.go]: nit:  a switch probably reads better

File: pkg/controllers/placement/namespace_status_sync.go [hunk 29]
```diff
@@ -27,81 +27,268 @@ import (
...
+	// Extract target namespace from resource selectors.
+	targetNamespace := extractNamespaceFromResourceSelectors(crp)
 	if targetNamespace == "" {
-		// Not NamespaceAccessible or no namespace found - skip sync.
-		klog.V(2).InfoS("Skipped processing CRP to create/update ClusterResourcePlacementStatus", "crp", klog.KObj(crp))
-		return nil
+		klog.ErrorS(controller.NewUnexpectedBehaviorError(fmt.Errorf("no namespace selector found despite NamespaceAccessible scope")),
+			"Failed to find valid Namespace selector for NamespaceAccessible scope",
+			"crp", placementKObj)
+		return "", nil
```
Comment (ryanzhang-oss) [inline pkg/controllers/placement/namespace_status_sync.go]: Nit: The fact that we are not returning error here is a bit odd. I know the reason is that the buildNamespaceAccessibleCondition function takes in two bool (isSyncError, isResourceSelectorInvalid) but conceptually, it should  only take in one. The function already takes in the msg but choose to decide the reason field.  This awkward setup could use some streamline

**Review by ryanzhang-oss — APPROVED** [review]
